File: Data Structures & Algorithms/count-number-of-islands/submission-0.cpp
class Solution {
public:
int delrow[4]={-1,0,+1,0};
int delcol[4]={0,+1,0,-1};
void dfs(int r, int c, int n, int m, vector<vector<int>>&vis, vector<vector<char>>&grid){
    vis[r][c]=1;
    for(int i=0;i<4;i++){
        int nr=r+delrow[i];
        int nc=c+delcol[i];

        if(nr>=0 && nr<n && nc>=0 && nc<m && !vis[nr][nc] && grid[nr][nc]=='1'){
            // vis[nr][nc]=1;
            dfs(nr,nc,n,m,vis,grid);
        }
    }
}
    int numIslands(vector<vector<char>>& grid) {
        int n=grid.size();
        int m=grid[0].size();
        int count=0;

        vector<vector<int>>vis(n, vector<int>(m,0));
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                if(grid[i][j]=='1' && !vis[i][j]){
                    count++;
                    dfs(i,j,n,m,vis,grid);
                }
            }
        }

        return count;
    }
};

File: Data Structures & Algorithms/search-for-word/submission-0.cs
public class Solution {
    int []delrow={-1,0,+1,0};
    int [] delcol={0,+1,0,-1};
    bool dfs(int r, int c, int k, int n, int m, string word, int [][] vis, char[][] board){
        if(k==word.Length) return true;

        vis[r][c]=1;
        for(int i=0;i<4;i++){
            int nr=r+delrow[i];
            int nc=c+delcol[i];

            if(nr>=0 && nr<n && nc>=0 && nc<m && vis[nr][nc]==0 && board[nr][nc]==word[k]){
                if(dfs(nr,nc,k+1,n,m,word,vis,board)) return true;
            }
        }

        vis[r][c]=0;

        return false;

    }
    public bool Exist(char[][] board, string word) {
        int n=board.Length;
        int m=board[0].Length;

        int[][] vis=new int[n][];
        for(int i=0;i<n;i++){
            vis[i]=new int[m];
        }

        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                if(board[i][j]==word[0]){
                    if(dfs(i,j,1,n,m,word,vis,board)==true) return true;
                }
            }
        }

        return false;
    }
}
